cluster/ubuntu/util.sh

```diff
@@ -346,99 +349,229 @@ function kube-up() {
-  echo
+  
+  echo -e "\nDeploying master on machine ${MASTER_IP}"
+
   ssh $SSH_OPTS $MASTER "mkdir -p ~/kube/default"
-  scp -r $SSH_OPTS saltbase/salt/generate-cert/make-ca-cert.sh ubuntu/reconfDocker.sh ubuntu/config-default.sh ubuntu/util.sh ubuntu/master/* ubuntu/binaries/master/ "${MASTER}:~/kube"
-
-  # remote login to MASTER and use sudo to configue k8s master
-  ssh $SSH_OPTS -t $MASTER "source ~/kube/util.sh; \
-                            setClusterInfo; \
-                            create-etcd-opts; \
-                            create-kube-apiserver-opts "${SERVICE_CLUSTER_IP_RANGE}" "${ADMISSION_CONTROL}" "${SERVICE_NODE_PORT_RANGE}"; \
-                            create-kube-controller-manager-opts "${NODE_IPS}"; \
-                            create-kube-scheduler-opts; \
-                            create-flanneld-opts "127.0.0.1"; \
-                            sudo -p '[sudo] password to start master: ' cp ~/kube/default/* /etc/default/ && sudo cp ~/kube/init_conf/* /etc/init/ && sudo cp ~/kube/init_scripts/* /etc/init.d/ ;\
-                            sudo groupadd -f -r kube-cert; \
-                            ${PROXY_SETTING} sudo -E ~/kube/make-ca-cert.sh ${MASTER_IP} IP:${MASTER_IP},IP:${SERVICE_CLUSTER_IP_RANGE%.*}.1,DNS:kubernetes,DNS:kubernetes.default,DNS:kubernetes.default.svc,DNS:kubernetes.default.svc.cluster.local; \
-                            sudo mkdir -p /opt/bin/ && sudo cp ~/kube/master/* /opt/bin/; \
-                            sudo service etcd start; \
-                            sudo FLANNEL_NET=${FLANNEL_NET} ~/kube/reconfDocker.sh "a";" 
-}
+
+  # copy the binaries and scripts to the ~/kube directory on the master
+  scp -r $SSH_OPTS \
+    saltbase/salt/generate-cert/make-ca-cert.sh \
+    ubuntu/reconfDocker.sh \
+    ubuntu/config-default.sh \
+    ubuntu/util.sh \
+    ubuntu/master/* \
+    ubuntu/binaries/master/ \
+    "${MASTER}:~/kube"
+
+  EXTRA_SANS=(
+    IP:$MASTER_IP
+    IP:${SERVICE_CLUSTER_IP_RANGE%.*}.1
+    DNS:kubernetes
+    DNS:kubernetes.default
+    DNS:kubernetes.default.svc
+    DNS:kubernetes.default.svc.cluster.local
+  )
+  
+  EXTRA_SANS=$(echo ${EXTRA_SANS[@]} | tr ' ' ',')
+
+  # remote login to MASTER and configue k8s master
+  ssh $SSH_OPTS -t ${MASTER} "
+    source ~/kube/util.sh
+
+    setClusterInfo
+    create-etcd-opts
+    create-kube-apiserver-opts \
+      '${SERVICE_CLUSTER_IP_RANGE}' \
+      '${ADMISSION_CONTROL}' \
+      '${SERVICE_NODE_PORT_RANGE}'
+    create-kube-controller-manager-opts '${NODE_IPS}'
+    create-kube-scheduler-opts
+    create-flanneld-opts '127.0.0.1'
+    sudo -E -p '[sudo] password to start master: ' -- /bin/bash -c '
+      cp ~/kube/default/* /etc/default/ 
+      cp ~/kube/init_conf/* /etc/init/ 
+      cp ~/kube/init_scripts/* /etc/init.d/
+      
+      groupadd -f -r kube-cert
+      "${PROXY_SETTING}" ~/kube/make-ca-cert.sh "${MASTER_IP}" "${EXTRA_SANS}"
+      mkdir -p /opt/bin/
```

Comment:
@MikeSpreitzer Please check this 'cause i could barely catch what `squote` exactly means :-)